### PR TITLE
Add GCM documentation

### PR DESCRIPTION
**Pull request by vmarmol**

### CODE REVIEW

**Comment by vmarmol** [comment]
PTAL @vishh, went with separate config files to make the getting started easier.

File: deploy/kube-config/gcm/heapster-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  replicaSelector:
+    name: "heapster"
+  podTemplate:
+    desiredState:
+      manifest:
+        version: "v1beta1"
+        id: "monitoring-heapster-controller"
+        containers:
+          - name: "heapster"
+            image: "kubernetes/heapster:canary"
```
Comment (vmarmol) [inline deploy/kube-config/gcm/heapster-controller.yaml]: Note that we use the canary here until 0.14.0 is released :)

Comment (vishh) [inline deploy/kube-config/gcm/heapster-controller.yaml]: Ack.

File: deploy/kube.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/kube-config"
```
Comment (vishh) [inline deploy/kube.sh]: Shall we use this script to deploy heapster with different sinks? 
Ideally I want to have a go binary for deployment, but this script isn't a bad place to start.

File: docs/gcm.md [hunk 3]
```diff
@@ -1,3 +1,24 @@
 ## Running Heapster in Kubernetes with a Google Cloud Monitoring backend
 
-**Coming soon!**
+This guide assumes you are running on Google Compute Engine.
```
Comment (vishh) [inline docs/gcm.md]: Should we instead state that GCM will work only when running on GCE?

File: docs/gcm.md [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+
+### Setup a Kubernetes cluster
+[Bring up a Kubernetes cluster](https://github.com/GoogleCloudPlatform/kubernetes), if you haven't already. Ensure that `kubecfg.sh` is exported. By default, [cAdvisor](https://github.com/google/cadvisor) runs on all nodes in port 4194.
```
Comment (vishh) [inline docs/gcm.md]: We don't access cadvisor directly anymore in heapster. Should we even mention cadvisor port here?

File: docs/gcm.md [hunk 5]
```diff
@@ -1,3 +1,24 @@
...
+```
+
+Heapster metrics are now being exported to Google Cloud Monitoring as custom metrics.
+
+### Metrics Dashboard
+To access the Google Cloud Monitoring dashboard go to: [https://app.google.stackdriver.com/](https://app.google.stackdriver.com/). Create a new dashboard and add the desired charts. Select the *Custom Metric* Resource Type and all Heapster metrics are under the `kubernetes.io` namespace. You can narrow down the query by the metric labels provided.
+
+It is also possible to query the Google Cloud Monitoring data directly using their [custom metric read API](https://cloud.google.com/monitoring/v2beta2/timeseries/list).
+
+## Hints
```
Comment (vishh) [inline docs/gcm.md]: This should go in a central location. May be in the landing page?

**Comment by vishh** [comment]
Overall LGTM. Just a few comments. Should we wait until 0.10.0 is released?

File: deploy/kube.sh (hunk 2 above)
Comment (vmarmol) [inline deploy/kube.sh]: Plan SGTM, mind we do it in a followup PR?

File: docs/gcm.md (hunk 5 above)
Comment (vmarmol) [inline docs/gcm.md]: We should probably have it in a separate troubleshooting page. Seems weird to be in the landing page...Added to the debugging page. WDYT?

File: docs/gcm.md (hunk 4 above)
Comment (vmarmol) [inline docs/gcm.md]: Yes, removed (and in influxdb.md)

File: docs/gcm.md (hunk 3 above)
Comment (vmarmol) [inline docs/gcm.md]: Yes, done.

**Comment by vmarmol** [comment]
Thanks for the review @vishh! We can wait or push it before, either should be fine :)

File: deploy/kube.sh (hunk 2 above)
Comment (vishh) [inline deploy/kube.sh]: Sounds good.

File: deploy/kube-config/gcm/heapster-controller.yaml (hunk 1 above)
Comment (vishh) [inline deploy/kube-config/gcm/heapster-controller.yaml]: @vmarmol: Can you update this to v0.10.0 since its released?

**Comment by vishh** [comment]
Needs a rebase. Feel free to self merge once you have updated the release to v0.10.0 in the guide.

**Comment by vmarmol** [comment]
Rebased, waiting on green.

**Comment by vmarmol** [comment]
Green :D

**Pull request merged**